modules/lib/types.nix

```diff
@@ -0,0 +1,22 @@
+{ lib }:
+
+with lib;
+{
+  selectorFunction = mkOptionType {
+    name = "selectorFunction";
+    description =
+      "Function that takes an attribute set and returns a list " +
+      "containing a selection of the values of the input set";
+    check = select: isFunction select;
```

Comment:
Can simplify to `check = isFunction;`.

Reply:
Hah. I had this done for `overlayFunction` :+1: